plugin/src/main/kotlin/com/varabyte/kobweb/intellij/startup/KobwebPostStartupProjectActivity.kt

```diff
@@ -1,25 +1,106 @@
 package com.varabyte.kobweb.intellij.startup
 
+import com.intellij.notification.NotificationType
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.actionSystem.ex.ActionUtil
+import com.intellij.openapi.actionSystem.ex.AnActionListener
 import com.intellij.openapi.components.service
+import com.intellij.openapi.externalSystem.action.RefreshAllExternalProjectsAction
 import com.intellij.openapi.externalSystem.service.project.manage.ProjectDataImportListener
 import com.intellij.openapi.project.Project
+import com.intellij.openapi.project.modules
+import com.intellij.openapi.project.rootManager
+import com.intellij.openapi.roots.LibraryOrderEntry
+import com.intellij.openapi.roots.ModuleOrderEntry
 import com.intellij.openapi.startup.ProjectActivity
+import com.varabyte.kobweb.intellij.notification.KobwebNotificationHandle
+import com.varabyte.kobweb.intellij.notification.KobwebNotifier
 import com.varabyte.kobweb.intellij.services.project.KobwebProjectCacheService
+import com.varabyte.kobweb.intellij.util.kobweb.KobwebPluginState
+import com.varabyte.kobweb.intellij.util.kobweb.kobwebPluginState
+
+/**
+ * A simple heuristic for checking if this project has code in it somewhere that depends on Kobweb.
+ *
+ * This is useful as users may install the Kobweb plugin for one or two of their projects, but we should stay out of the
+ * way in every other kind of project.
+ */
+private fun Project.hasAnyKobwebDependency(): Boolean {
```

Comment:
The util package would probably be better suited for that function. We already have the KobwebProjectUtils and stuff there

Reply:
I'd rather keep it private for now, and move this over there only if we're ever sure it's needed anywhere else. I'm not sure I designed this generally enough to feel comfortable for people to use it anywhere they want to. (For example, I didn't time it carefully, and it's probably not something I want people using thoughtlessly inside Kobweb actions)

Reply:
Alright thats a good point